File: app/src/main/java/com/ruzhan/recyclerviewitemanimation/holder/OneSlideViewHolder.java
package com.ruzhan.recyclerviewitemanimation.holder;

import android.animation.Animator;
import android.util.Log;
import android.view.View;
import android.widget.TextView;

import com.ruzhan.recyclerviewitemanimation.R;

import zhan.library.slide.SlideAnimatorListener;
import zhan.library.slide.helper.SlideAnimationHelper;
import zhan.library.slide.holder.SlideViewHolder;


public class OneSlideViewHolder extends SlideViewHolder {

    private static final String TAG = OneSlideViewHolder.class.getSimpleName();

    private View mContentRl;
    private TextView itemTv;

    private View itemIconV;
    private View titleLl;
    private TextView itemTitleTv;
    private TextView itemSubtitleTv;

    public OneSlideViewHolder(View itemView) {
        super(itemView);

        mContentRl = itemView.findViewById(R.id.item_content_rl);
        itemTv = (TextView) itemView.findViewById(R.id.item_tv);
        itemIconV = itemView.findViewById(R.id.item_icon_v);
        titleLl = itemView.findViewById(R.id.title_ll);
        itemTitleTv = (TextView) itemView.findViewById(R.id.item_title_tv);
        itemSubtitleTv = (TextView) itemView.findViewById(R.id.item_subtitle_tv);

        setSlideAnimatorListener(new SlideAnimatorListener() {
            @Override
            public void onSlideAnimationStart(Animator animation, int currentStatus) {
                Log.i(TAG, "setSlideAnimatorListener: onSlideAnimationStart: "
                        + (currentStatus == SlideAnimationHelper.STATE_OPEN ? "open" : "close"));
            }

            @Override
            public void onSlideAnimationEnd(Animator animation, int currentStatus) {
                Log.i(TAG, "setSlideAnimatorListener: onSlideAnimationEnd: "
                        + (currentStatus == SlideAnimationHelper.STATE_OPEN ? "open" : "close"));
            }
        });
    }

    @Override
    public void doAnimationSet(int offset, float fraction) {
        mContentRl.scrollTo(offset, 0);

        itemTv.setScaleX(fraction);
        itemTv.setScaleY(fraction);
        itemTv.setAlpha(fraction * 255);

        titleLl.scrollTo(offset, 0);
    }

    @Override
    public void onBindSlideClose(int state) {
        titleLl.scrollTo(0, 0);
    }

    @Override
    public void doAnimationSetOpen(int state) {
        titleLl.scrollTo(-mOffset, 0);
    }

    public void bind() {
        setOffset(50);

        //slide must call
        onBindSlide(mContentRl);
    }
}

File: app/src/main/java/com/ruzhan/recyclerviewitemanimation/holder/OneSlideViewHolder.kt
package com.ruzhan.recyclerviewitemanimation.holder

import android.animation.Animator
import android.util.Log
import android.view.View
import kotlinx.android.synthetic.main.one_item.view.*
import zhan.library.slide.SlideAnimatorListener
import zhan.library.slide.helper.SlideAnimationHelper
import zhan.library.slide.holder.SlideViewHolder


class OneSlideViewHolder(itemView: View) : SlideViewHolder(itemView) {

    companion object {
        private const val TAG = "OneSlideViewHolder"
        private const val DEFAULT_OFFSET = 50
    }

    init {
        setSlideAnimatorListener(object : SlideAnimatorListener() {
            override fun onSlideAnimationStart(animation: Animator, currentStatus: Int) {
                Log.i(TAG, "setSlideAnimatorListener: onSlideAnimationStart: " +
                        if (currentStatus == SlideAnimationHelper.STATE_OPEN) "open" else "close")
            }

            override fun onSlideAnimationEnd(animation: Animator, currentStatus: Int) {
                Log.i(TAG, "setSlideAnimatorListener: onSlideAnimationEnd: " +
                        if (currentStatus == SlideAnimationHelper.STATE_OPEN) "open" else "close")
            }
        })
    }

    override fun doAnimationSet(offset: Int, fraction: Float) {
        itemView.item_content_rl.scrollTo(offset, 0)

        itemView.item_tv.scaleX = fraction
        itemView.item_tv.scaleY = fraction
        itemView.item_tv.alpha = fraction * 255
        itemView.title_ll.scrollTo(offset, 0)
    }

    override fun onBindSlideClose(state: Int) {
        itemView.title_ll.scrollTo(0, 0)
    }

    override fun doAnimationSetOpen(state: Int) {
        itemView.title_ll.scrollTo(-mOffset, 0)
    }

    fun bind() {
        offset = DEFAULT_OFFSET
        //slide must call
        onBindSlide(itemView.item_content_rl)
    }
}
